packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -445,10 +446,65 @@ class _BoxDecorationPainter extends BoxPainter {
 
   void _paintBackgroundColor(Canvas canvas, Rect rect, TextDirection? textDirection) {
     if (_decoration.color != null || _decoration.gradient != null) {
-      _paintBox(canvas, rect, _getBackgroundPaint(rect, textDirection), textDirection);
+      // When border is filled, the rect is reduced to avoid anti-aliasing
+      // rounding error leaking the background color around the clipped
+      // shape.
+      // https://github.com/flutter/flutter/issues/13675
+      final Rect ajustedRect = _adjustRectOnOutlinedBorder(rect, textDirection);
```

Comment:
nit: `adjusted`

Reply:
Thanks! I'm traveling, I'll fix this (and my other PRs) next week.